src/tile/MediaView.test.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { describe, expect, test } from "vitest";
+import { render, screen } from "@testing-library/react";
+import { axe } from "vitest-axe";
+import { TooltipProvider } from "@vector-im/compound-web";
+import {
+  TrackReference,
+  TrackReferencePlaceholder,
+} from "@livekit/components-core";
+import { Track, TrackPublication } from "livekit-client";
+
+import { MediaView } from "./MediaView";
+import { EncryptionStatus } from "../state/MediaViewModel";
+import { mockLocalParticipant } from "../utils/test";
+
+describe("MediaView", () => {
+  const participant = mockLocalParticipant({});
+  const trackReferencePlaceholder: TrackReferencePlaceholder = {
+    participant,
+    source: Track.Source.Camera,
+  };
+  const trackReference: TrackReference = {
+    ...trackReferencePlaceholder,
+    publication: new TrackPublication(Track.Kind.Video, "id", "name"),
+  };
+
+  test("is accessible", async () => {
+    const { container } = render(
+      <MediaView
+        displayName="Bob"
+        videoEnabled
+        videoFit="contain"
+        targetWidth={300}
+        targetHeight={200}
+        encryptionStatus={EncryptionStatus.Connecting}
+        mirror={false}
+        unencryptedWarning={false}
+        video={trackReference}
+        member={undefined}
+      />,
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  describe("placeholder track", () => {
+    test("neither video nor avatar are shown", () => {
+      render(
+        <MediaView
+          displayName="Bob"
+          videoEnabled
+          videoFit="contain"
+          targetWidth={300}
+          targetHeight={200}
+          encryptionStatus={EncryptionStatus.Connecting}
+          mirror={false}
+          unencryptedWarning={false}
+          video={trackReferencePlaceholder}
+          member={undefined}
+        />,
+      );
+      expect(screen.queryByTestId("video")).toBeNull();
+      expect(screen.getByTestId("avatar")).not.toBeVisible();
+    });
+  });
+
+  describe("name tag", () => {
+    test("is shown", () => {
+      render(
+        <MediaView
+          displayName="Bob"
+          videoEnabled
+          videoFit="contain"
+          targetWidth={300}
+          targetHeight={200}
+          encryptionStatus={EncryptionStatus.Connecting}
+          mirror={false}
+          unencryptedWarning={false}
+          video={trackReference}
+          member={undefined}
+        />,
+      );
+      expect(screen.getByTestId("name_tag")).toHaveTextContent("Bob");
+    });
+  });
```

Comment:
Can we not combine those two tests?

Reply:
Whilst the code is very similar and the expectations could be combined, I would argue that the purpose of each test case is clearer in the way that it is currently written.

If you are concerned about code duplication I think I can do something about that whilst keeping things separate. I will push an extra commit.